extensions-builtin/canvas-zoom-and-pan/javascript/zoom.js

```diff
@@ -42,37 +42,124 @@ onUiLoaded(async() => {
         return (
-            typeof value === "string" && value.length === 1 && /[a-z]/i.test(value)
+            (typeof value === "string" &&
+                value.length === 1 &&
+                /[a-z]/i.test(value)) ||
+            specialKeys.includes(value)
         );
     }
 
     // Create hotkeyConfig from opts
     function createHotkeyConfig(defaultHotkeysConfig, hotkeysConfigOpts) {
         const result = {};
-        const usedKeys = new Set();
+        const usedSingleKeys = new Set();
+        const usedSpecialKeys = new Set();
+
+        // Normalize hotkey
+        function normalizeHotkey(hotkey) {
+            return hotkey.length === 1 ? "Key" + hotkey.toUpperCase() : hotkey;
+        }
+
+        // Format hotkey for display
+        function formatHotkeyForDisplay(hotkey) {
+            return hotkey.startsWith("Key") ? hotkey.slice(3) : hotkey;
+        }
+
+        // Check if canvas_hotkey_adjust and canvas_hotkey_zoom are the same
+        if (
+            hotkeysConfigOpts.canvas_hotkey_adjust !== "Disable" &&
+            hotkeysConfigOpts.canvas_hotkey_zoom !== "Disable" &&
+            normalizeHotkey(hotkeysConfigOpts.canvas_hotkey_adjust) ===
+                normalizeHotkey(hotkeysConfigOpts.canvas_hotkey_zoom)
+        ) {
+            console.error(
+                `Hotkey: ${formatHotkeyForDisplay(
+                    hotkeysConfigOpts.canvas_hotkey_zoom
+                )} for canvas_hotkey_zoom conflicts with canvas_hotkey_adjust. The default hotkey is used: ${formatHotkeyForDisplay(
+                    defaultHotkeysConfig.canvas_hotkey_zoom
+                )}`
+            );
+            hotkeysConfigOpts.canvas_hotkey_zoom =
+                defaultHotkeysConfig.canvas_hotkey_zoom;
+        }
 
         for (const key in defaultHotkeysConfig) {
             if (typeof hotkeysConfigOpts[key] === "boolean") {
                 result[key] = hotkeysConfigOpts[key];
                 continue;
             }
+            if (hotkeysConfigOpts[key] === "Disable") {
+                result[key] = hotkeysConfigOpts[key];
+                continue;
+            }
             if (
                 hotkeysConfigOpts[key] &&
-                isSingleLetter(hotkeysConfigOpts[key]) &&
-                !usedKeys.has(hotkeysConfigOpts[key].toUpperCase())
+                isValidHotkey(hotkeysConfigOpts[key])
             ) {
-                // If the property passed the test and has not yet been used, add 'Key' before it and save it
-                result[key] = "Key" + hotkeysConfigOpts[key].toUpperCase();
-                usedKeys.add(hotkeysConfigOpts[key].toUpperCase());
+                const hotkey = normalizeHotkey(hotkeysConfigOpts[key]);
+                const isSpecialKey = hotkey.length > 1;
+
+                if (
+                    (!isSpecialKey && !usedSingleKeys.has(hotkey)) ||
+                    (isSpecialKey && !usedSpecialKeys.has(hotkey))
+                ) {
```

Comment:
It can be simpler of the 'if()'

Reply:
Yes, I'll fix the code and make it more compact in the coming days, for now it just works.

Reply:
Corrected, removed this piece of code and made it smaller and clearer